nucypher/blockchain/eth/signers/hardware.py

```diff
@@ -0,0 +1,243 @@
+        Derive trezor addresses up to ADDRESS_CACHE_SIZE relative to
+        the calculated base path and internally cache them.
+        """
+        for index in range(self.ADDRESS_CACHE_SIZE):
+            hd_path = self.__get_address_path(index=index)
+            address = self.__get_address(hd_path=hd_path, show_display=False)
+            self.__addresses[address] = hd_path
+
+    @staticmethod
+    def _format_transaction(transaction_dict: dict) -> dict:
+        """
+        Handle Web3.py -> Trezor native transaction field formatting
+        # https://web3py.readthedocs.io/en/latest/web3.eth.html#web3.eth.Eth.sendRawTransaction
+        """
+        assert_valid_fields(transaction_dict)
+        trezor_transaction_keys = {'gas': 'gas_limit', 'gasPrice': 'gas_price', 'chainId': 'chain_id'}
+        trezor_transaction = dict(apply_key_map(trezor_transaction_keys, transaction_dict))
+        return trezor_transaction
+
+    @handle_trezor_call
+    def __sign_transaction(self, n: List[int], trezor_transaction: dict) -> Tuple[bytes, bytes, bytes]:
+        """Internal wrapper for trezorlib transaction signing calls"""
+        v, r, s = trezorlib.ethereum.sign_tx(client=self.__client, n=n, **trezor_transaction)
+        return v, r, s
+
+    #
+    # Trezor Signer API
+    #
+
+    @classmethod
+    def from_signer_uri(cls, uri: str) -> 'TrezorSigner':
+        """Return a trezor signer from URI string i.e. trezor:///my/trezor/path """
+        if uri != cls.URI_SCHEME:
+            raise cls.InvalidSignerURI(f'{uri} is not a valid trezor URI scheme')
+        return cls()
+
+    def is_device(self, account: str) -> bool:
+        """Trezor is always a device."""
+        return True
+
+    @validate_checksum_address
+    def unlock_account(self, account: str, password: str, duration: int = None) -> bool:
+        """Defer account unlocking to the trezor, do not indicate application level unlocking logic."""
+        return True
+
+    @validate_checksum_address
+    def lock_account(self, account: str) -> bool:
+        """Defer account locking to the trezor, do not indicate application level unlocking logic."""
+        return True
+
+    @property
+    def accounts(self) -> List[str]:
+        """Returns a list of cached trezor checksum addresses from initial derivation."""
+        return list(self.__addresses.keys())
+
+    @handle_trezor_call
+    def sign_message(self, message: bytes, checksum_address: str) -> SignedMessage:
+        """
+        Signs a message via the TREZOR ethereum sign_message API and returns
+        a named tuple containing the signature and the address used to sign it.
+        This method requires interaction between the TREZOR and the user.
+        """
+        hd_path = self.__get_address_path(checksum_address=checksum_address)
+        signed_message = trezorlib.ethereum.sign_message(self.__client, hd_path, message)
+        return self.SignedMessage(signed_message.signature, signed_message.address)
```

Comment:
This seems to be breaking the expected return type in the abstract class, as well as wrt the behavior of other signers:
```
@abstractmethod
    def sign_message(self, account: str, message: bytes, **kwargs) -> HexBytes:
        return NotImplemented
```

Reply:
Opened new Issues #2262 and #2261